plugins/http/guest-js/index.ts

```diff
@@ -155,11 +155,11 @@ export async function fetch(
     rid,
   });
 
-  const body = await invoke<number[]>("plugin:http|fetch_read_body", {
+  const body = await invoke<ArrayBuffer>("plugin:http|fetch_read_body", {
```

Comment:
invoke is not guaranteed to return an `ArrayBuffer`, it will only return `ArrayBuffer` if using the v2 IPC protocol through `fetch`, if it used v1 IPC (used on Linux and sometimes on macOS (remote urls)), it will be an array. 
```suggestion
  const body = await invoke<ArrayBuffer | number[]>("plugin:http|fetch_read_body", {
```

and so later we need to check if it is a normal array or array buffer

```js
(body instanceof ArrayBuffer && body.byteLength ) || (body instanceof Array && body.length > 0) ? body : null
```